ddtrace/internal/flare.py

```diff
@@ -29,13 +27,28 @@
 log = get_logger(__name__)
 
 
+class FlarePrepRequest:
+    def __init__(self, log_level: str):
+        if logging.getLevelName(log_level) == f"Level {log_level}":
+            raise TypeError("Invalid log level provided: %s", log_level)
+        self.log_level = log_level
```

Comment:
do you want to just store this as the integer rather than the name?

you could do something like this:

```
log_level: int

def __init__(self, log_level_name: str):
    log_level: int | str = logging.getLevelName(log_level_name)
    if not isinstance(log_level, int):
        raise ValueError(f"Unknown log level: {log_level_name}")
    self.log_level = log_level
```